### ISSUE TITLE
Different results with and without installed Git

### ISSUE THREAD

**Issue by beatcracker** [editorconfig-checker/editorconfig-checker#261]
I was trying to understand why I get different results while running `ec` in the container with mounted volume vs on host directly. In particular, it was checking binary files in the `.git` folder. 

The culprit seems to be that under the hood `ec` is invoking Git to get a list of files in the repo:

- https://github.com/editorconfig-checker/editorconfig-checker/blob/e2ce8ab5f3dd42a9b93a3ff2001ce007bea885c2/pkg/files/files.go#L98

This is not mentioned anywhere in the docs and no warning is issued when Git binary is not available or the command fails for other reasons.

Ideally, this has to be configurable. Something like `-no-git` option to preserve backward compatibility + a warning in case the Git support is enabled but fails.

I might be willing to take a stab at this if you're ok with the idea 😺 .

**Comment by polarathene** [comment]
> Something like `-no-git` option to preserve backward compatibility + a warning in case the Git support is enabled but fails.

`git` has been present in the project [since Oct 2018 with version `0.0.1`](https://github.com/editorconfig-checker/editorconfig-checker/commit/c718a75b4bfc1b884245be1f1ae6da8147bd0f92#diff-9e185f29fa355d7dd8fdd9c9ff1d0723b85206aa7d37c4eec93997005dc291ebR92). Thus not really a backward compatibility concern.

If you look at the lines after your referenced one (_which has an `err`_), you'll see that it fallsback to handling when `git` command fails:

https://github.com/editorconfig-checker/editorconfig-checker/blob/e2ce8ab5f3dd42a9b93a3ff2001ce007bea885c2/pkg/files/files.go#L98-L104

---

You did not state if `git` was not installed on your host, but it should be clear it was in the Docker image, where your problem was presumably happening?

A newer version of `git` was released at some point that defaulted to some safety check which was breaking the Docker image functionality. As a result there was no implicit ignoring of `.git`. [This was resolved in version `2.7.1`](https://github.com/editorconfig-checker/editorconfig-checker/issues/268#issuecomment-1747736220).

On a host where `git` is not available, I imagine it'll handle the fallback scenario where it is unable to infer the git repo files under `.git/` should be ignored. For that you'd want to configure an exclusion pattern, either via `.ecrc.json` config or the CLI `--exclude` option.

[Here's an example](https://github.com/docker-mailserver/docker-mailserver/blob/aae42fae9bd0190c7b4326e3855d8492ecd61dfa/test/linting/.ecrc.json), note how you need to escape the `.` as it's a regex pattern, so `\\.git/` will do the trick.

**Comment by nikolay** [comment]
@polarathene Is `.ecrc.json` supported as your documentation does not mention it as an alternative to `.ecrc`?

**Comment by polarathene** [comment]
> your documentation does not mention it as an alternative to `.ecrc`?

I am only marked as a contributor because I had a PR updating the README merged. Not my documentation? 😅 

The project I am active on is `docker-mailserver`, we've been using `.ecrc.json` since it is json config but [we explicitly provide it to the `-config` option](https://github.com/docker-mailserver/docker-mailserver/blob/d988d8a8d1be8bd07c271eaa9911f15f5cd033c7/test/linting/lint.sh#L20-L24), which would let you use any filename.

So no, it's probably not an officially supported filename 🤷‍♂️ Easy enough for you to check though 👍

**Comment by nikolay** [comment]
@polarathene Sorry for this! I just opened an [issue](https://github.com/editorconfig-checker/editorconfig-checker/issues/306). Please, upvote if you believe it will add value to the project. Thanks!

**Comment by rasa** [comment]
Fixed in #427, so this ticket can be closed.

**Comment by klaernie** [comment]
Good point, closing the issue, since the problem was only exactly `.git`.
